cine-flex/src/components/ShortTeasersMovie/ShortTeasersMovie.jsx

```diff
@@ -0,0 +1,123 @@
+import styles from '../ShortTeasersMovie/ShortTeasersMovie.module.scss';
+import { AppConstants } from '../../constants/app-constants';
+import React, { useState, useRef, useEffect } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faPlay, faPause } from '@fortawesome/free-solid-svg-icons';
+import withAd from '../HOC/withAd';
+
+/**
+ * @description This function returns a teaser component present in short teasers section of home page
+ * @param {*} posterImg - string
+ * @param {*} movieName - string
+ * @returns 
+ */
+function ShortTeasersMovie(props) {
+    const { posterImg, movieName, isShowingAd, selectedAd, decrementTimeLeftForAdStart, pauseTimer, secsLeftForAdStart, secsLeftForAdStop, isAdEnded } = props;
+    const videoRef = useRef(null);
+    const [isPlaying, setIsPlaying] = useState(false);
+    let adImg;
+
+    useEffect(() => {
+        if(isShowingAd) {
+            videoRef.current.pause();
+            setIsPlaying(false);
+        }
+    },[isShowingAd]);
+
+    useEffect(() => {
+        if(isAdEnded) {
+            const isAnyOtherVideoPlaying = areOtherVideosPlaying();
```

Comment:
could you let me know why are we checking whether other videos are played or not ?

Reply:
I have made 2 or more videos not be played simultaneously for user experience @DurgaBNVL

Reply:
ok, is this feature implemented by discussing with trainer in terms of understanding requirements ?
or you have implemented it before discussing ?